### DISCUSSION TITLE
Sonar Transducer Shell (Abyssal Creature Tracker, Orientational Beacon..)

### DISCUSSION THREAD

**Discussion by Smrtonosan** [FakeFishGames/Barotrauma#12432] [Ideas and suggestions]
Specialized railgun shell that once fired from a railgun will start emitting a noise that can be seen on a sonar monitor as a marker.
- _Example of a renamed sonar marker as seen on the Navigation Terminal (From an active Sonar Beacon inside a Railgun Shell)_
![20230728203924_1](https://github.com/Regalis11/Barotrauma/assets/104994469/c583efa4-d11f-40a7-8cc3-38e410a525e3)

- _Shell's marker can be renamed by holding the shell in both hands, same as the handheld Sonar Beacon item._

![Rename](https://github.com/Regalis11/Barotrauma/assets/104994469/2ee99c44-04b3-4a61-b1ef-05ca69151904)

- _The shell requires a battery power to operate, if fired without a battery it will not emit a sound, nor create a marker on the sonar._

![20230728223839_1](https://github.com/Regalis11/Barotrauma/assets/104994469/5979911a-844a-4290-bae5-2dc5506fed8e)

What is written above can already be done with a Sonar Beacon inside a Railgun Shell, where the specialized item is different:

- Instead of exploding on contact, the Sonar Transducer Shell will embed itself into the surface or creature it hits, similar to harpoons.
- This can be used to track the position of abyssal creatures and enemy submarines. (Renowned for going out of sonar range.)
- This can also be used to mark a position on the sonar.
- This can also be used to pull creatures to the position/target you hit.

While there are many ways that it could be made to stick to targets, such as retractable barbs, I would propose that it instead used a malleable squash head made out of copper, a mostly underused metal. to stick to surfaces.

![Sonar Transducer Shell 2](https://github.com/Regalis11/Barotrauma/assets/104994469/df02c7e8-57ec-466c-a7aa-89aa950bec20)

Couple of issues I can think of and my solutions:

- Attacking smaller creatures with the shell such as Mudraptors, Hammerhead Spawn, Crawlers...
1. Shell will punch through the smaller creature with it's velocity, applying stun and some blunt force trauma, after either a grace period, or after it loses enough velocity creatures will be able to destroy it by attacking it, like the decoy depth charge.

- Sonar Transducer embeds itself into a larger pack(?) creature such as a Hammerhead.
1. Instead of hammerheads attacking the hammerhead with the transducer embedded into it, they will attack the transducer and then continue on with their business.

- Sonar Transducer is embedded into your submarine and you want to remove it.
1. You can interact with the embedded Sonar Transducer, allowing you to pull out the battery, disabling the beacon.
2. The embedded Sonar Transducer will disappear after the mission has ended.
3. The Sonar Transducer can be cut off the submarine with a Plasma Cutter, either destroying it with some metal breaking particle effects, or cutting it off will salvage the Sonar Transducer, allowing it to be brought back into the submarine and have it's copper squash head replaced, similar to the railgun's canister shot, this salvaging could also be done with Transducers embedded into larger creatures.

![Empty Sonar Transducer](https://github.com/Regalis11/Barotrauma/assets/104994469/1734ff10-f472-4e51-b93d-c7ee77ddd646)
![Canister 2](https://github.com/Regalis11/Barotrauma/assets/104994469/64f8bbb9-b652-403a-aa72-4b83a93ab4cb)
![Canister 1](https://github.com/Regalis11/Barotrauma/assets/104994469/83b319d6-c8fe-48d6-8c83-7eec2f187b1d)

Lastly, I would suggest that the Sonar Transducer Shell item becomes a Engineer talent item, requiring electrical skill to craft.

**Comment by Hazmat616** [comment]
- [ ] While I like this idea I'd love for more functionality be added to putting items inside the shells to alter functionality. So adding a beacon does all this or adding various meds or poisons cause clouds to form or be injected into larger creatures.
